src/erlcloud_aws.erl

```diff
@@ -266,21 +266,26 @@ format_timestamp({{Yr, Mo, Da}, {H, M, S}}) ->
 
 default_config() ->
     case get(aws_config) of
-        undefined ->
-            AccessKeyId = case os:getenv("AWS_ACCESS_KEY_ID") of
-                              false -> "";
-                              AKI -> AKI
-                          end,
-            SecretAccessKey = case os:getenv("AWS_SECRET_ACCESS_KEY") of
-                                  false -> "";
-                                  SAC -> SAC
-                              end,
+        undefined -> default_config_env();
+        Config -> Config
+    end.
+
+default_config_env() ->
+    case {os:getenv("AWS_ACCESS_KEY_ID"), os:getenv("AWS_SECRET_ACCESS_KEY")} of
+        {false, false} ->
```

Comment:
I don't see the point of using keys if one of them is not set. 
I'd say if both `{AccessKeyId, SecretAccessKey}` are not false then use them, otherwise use profile.